src/NuGet.Core/NuGet.Packaging.Core/NuspecUtility.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace NuGet.Packaging.Core
+{
+    /// <summary>
+    /// Until NuspecReader and Manifest are unified, this is a place to share implementations of
+    /// reading and parsing specific elements out of the .nuspec XML.
+    /// </summary>
+    public static class NuspecUtility
+    {
+        public static readonly string PackageTypes = "packageTypes";
+        public static readonly string PackageType = "packageType";
+        public static readonly string PackageTypeName = "name";
+        public static readonly string PackageTypeVersion = "version";
+
+        /// <summary>
+        /// Gets zero or more package types from the .nuspec.
+        /// </summary>
+        public static IReadOnlyList<PackageType> GetPackageTypes(XElement metadataNode, bool useMetadataNamespace)
+        {
+            IEnumerable<XElement> nodes;
+            if (useMetadataNamespace)
+            {
+                var metadataNamespace = metadataNode.GetDefaultNamespace().NamespaceName;
+                nodes = metadataNode
+                    .Elements(XName.Get(PackageTypes, metadataNamespace))
+                    .SelectMany(x => x.Elements(XName.Get(PackageType, metadataNamespace)));
+            }
+            else
+            {
+                nodes = metadataNode
+                    .Elements()
+                    .Where(x => x.Name.LocalName == PackageTypes)
```

Comment:
I did more investigation and I found that we treat tag name as case sensitive (which aligns with XML in general).